Cue/app/tabs/library/LibraryHome.js

```diff
@@ -125,6 +126,44 @@ class LibraryHome extends React.Component {
   }
 
   _onPressAddDeck = () => {
+    Alert.alert(
+      Platform.OS === 'android' ? 'Add deck' : 'Add Deck',
+      'You can create a new private deck or add a shared deck by entering a share code.',
+      [
+        {text: 'Cancel', style: 'cancel'},
+        {text: 'Look up share code', onPress: this._onPressLookUpDeckByShareCode},
+        {text: 'Create new deck', onPress: this._onPressCreateDeck},
+      ]
+    )
+  }
+
+  _onPressLookUpDeckByShareCode = () => {
+    CuePrompt.prompt(
+      Platform.OS === 'android' ? 'Look up share code' : 'Look Up Share Code',
+      '',
+      [
+        {text: 'Cancel', style: 'cancel'},
+        {text: 'Look up', onPress: this._onEnterShareCode},
+      ]
+    )
+  }
+
+  _onEnterShareCode = (shareCode: string) => {
+    if (shareCode && shareCode.length) {
+      LibraryApi.getUuidByShareCode(shareCode).then(json => {
+        LibraryApi.fetchDeck(json.uuid, shareCode).then(deck => {
```

Comment:
I think Henry might be right... If I return it then it becomes part of the chain